src/Microsoft.Kusto.ServiceLayer/DataSource/KustoClient.cs

```diff
@@ -173,22 +173,29 @@ public IDataReader ExecuteQuery(string query, CancellationToken cancellationToke
             clientRequestProperties.SetOption(ClientRequestProperties.OptionNoTruncation, true);
             cancellationToken.Register(() => CancelQuery(clientRequestProperties.ClientRequestId));
 
-            var kustoCodeService = new KustoCodeService(query);
-            var minimalQuery = kustoCodeService.GetMinimalText(MinimalTextKind.RemoveLeadingWhitespaceAndComments);
+            var script = CodeScript.From(query, GlobalState.Default);
+            List<IDataReader> origReaders = new List<IDataReader>();
 
-            try
+            foreach (var codeBlock in script.Blocks)
```

Comment:
Instead of going sequential, we should do this in parallel. Send all queries in parallel and wait for all to return.

Reply:
When doing in parallel, you might have to make sure that the results are put back in order. So using an array for the number of results might work out better. Each result can go into its query's index.